src/R/Editor/Impl/Completion/Definitions/RCompletionContext.cs

```diff
@@ -9,108 +9,84 @@
         public int Position { get; set; }
         public ICompletionSession Session { get; private set; }
         public ITextBuffer TextBuffer { get; private set; }
         public AstRoot AstRoot { get; private set; }
         public bool InternalFunctions { get; internal set; }
 
-        public RCompletionContext(ICompletionSession session, ITextBuffer textBuffer, AstRoot ast, int position)
-        {
+        public RCompletionContext(ICompletionSession session, ITextBuffer textBuffer, AstRoot ast, int position) {
             Session = session;
             TextBuffer = textBuffer;
             Position = position;
             AstRoot = ast;
         }
 
-        public bool IsInNameSpace()
-        {
+        public bool IsInNameSpace() {
             return IsCaretInNamespace(Session.TextView);
         }
 
-        public static bool IsCaretInNamespace(ITextView textView)
-        {
+        public static bool IsCaretInNamespace(ITextView textView) {
             SnapshotPoint? bufferPosition = REditorDocument.MapCaretPositionFromView(textView);
-            if (bufferPosition.HasValue)
-            {
+            if (bufferPosition.HasValue) {
                 return IsInNamespace(bufferPosition.Value.Snapshot, bufferPosition.Value.Position);
             }
 
             return false;
         }
 
-        public static bool IsInNamespace(ITextSnapshot snapshot, int position)
-        {
-            try
-            {
-                ITextSnapshotLine line = snapshot.GetLineFromPosition(position);
-                if (line.Length > 2 && position - line.Start > 2)
-                {
-                    return snapshot[position - 1] == ':';
-                }
+        public static bool IsInNamespace(ITextSnapshot snapshot, int position) {
+            ITextSnapshotLine line = snapshot.GetLineFromPosition(position);
+            if (line.Length > 2 && position - line.Start > 2) {
+                return snapshot[position - 1] == ':';
             }
-            catch (Exception) { }
 
             return false;
         }
 
-        public static bool IsCaretInLibraryStatement(ITextView textView)
-        {
-            try
-            {
+        public static bool IsCaretInLibraryStatement(ITextView textView) {
+            try {
                 SnapshotPoint? bufferPosition = REditorDocument.MapCaretPositionFromView(textView);
-                if (bufferPosition.HasValue)
-                {
+                if (bufferPosition.HasValue) {
                     ITextSnapshot snapshot = bufferPosition.Value.Snapshot;
                     int caretPosition = bufferPosition.Value.Position;
                     ITextSnapshotLine line = snapshot.GetLineFromPosition(caretPosition);
 
-                    if (line.Length < 8 || caretPosition < line.Start + 8 || snapshot[caretPosition - 1] != '(')
-                    {
+                    if (line.Length < 8 || caretPosition < line.Start + 8 || snapshot[caretPosition - 1] != '(') {
```

Comment:
Possibly another one to file for later - looks like this won't handle something like:
```r
library(
  foo
```